component/prometheus/exporter/exporter.go

```diff
@@ -134,6 +107,46 @@ func (c *Component) Update(args component.Arguments) error {
 	return err
 }
 
+// Handler serves metrics endpoint from the integration implementation.
+func (c *Component) Handler() http.Handler {
+	c.mut.Lock()
+	defer c.mut.Unlock()
+	return c.metricsHandler
+}
+
+func newExporter(creator Creator, name string, instance string, multiTargetFunc func(discovery.Target, component.Arguments) []discovery.Target) func(component.Options, component.Arguments) (component.Component, error) {
```

Comment:
this injection of instance is not going to work, as we have not evaluated the arguments yet :thinking:

Reply:
I will try tweakaing the `multiTargetFunc` to turn it into a more generic custom target generator that also allows overriding the instance label, as discussed with @captncraig and @spartan0x117